File: paiguangguang/backend/app/schemas/architecture.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ArchitectureGraphPosition(BaseModel):
    x: int
    y: int


class ArchitectureGraphNodeData(BaseModel):
    title: str = Field(min_length=1)
    type: str = Field(min_length=1)
    phase: str = Field(min_length=1)
    description: str = Field(min_length=1)


class ArchitectureGraphNode(BaseModel):
    id: str = Field(min_length=1)
    type: str = "default"
    position: ArchitectureGraphPosition
    data: ArchitectureGraphNodeData


class ArchitectureGraphEdge(BaseModel):
    id: str = Field(min_length=1)
    source: str = Field(min_length=1)
    target: str = Field(min_length=1)
    type: str = "smoothstep"
    animated: bool = False


class ArchitectureGraphData(BaseModel):
    nodes: list[ArchitectureGraphNode]
    edges: list[ArchitectureGraphEdge]

File: paiguangguang/frontend/types/architecture.ts
export type ArchitectureGraphPosition = {
  x: number;
  y: number;
};

export type ArchitectureGraphNodeData = {
  title: string;
  type: string;
  phase: string;
  description: string;
};

export type ArchitectureGraphNode = {
  id: string;
  type: string;
  position: ArchitectureGraphPosition;
  data: ArchitectureGraphNodeData;
};

export type ArchitectureGraphEdge = {
  id: string;
  source: string;
  target: string;
  type: string;
  animated: boolean;
};

export type ArchitectureGraphData = {
  nodes: ArchitectureGraphNode[];
  edges: ArchitectureGraphEdge[];
};
